src/test/s3/commands/uploadFile.test.ts

```diff
@@ -58,31 +197,182 @@ describe('uploadFileCommand', function () {
+        when(s3.uploadFile(anything())).thenReject(new Error('Expected failure'))
 
-        assert.strictEqual(commands.command, 'aws.refreshAwsExplorerNode')
-        assert.deepStrictEqual(commands.args, [node])
+        
+        window = new FakeWindow({ dialog: { openSelections: [fileLocation] } })
+        outputChannel = new MockOutputChannel()
+        await uploadFileCommand(instance(s3), fileLocation, statFile, getBucket, getFile, window, outputChannel, commands)
+         
+        assert.ok(window.message.error?.includes('Failed to upload file'))
     })
+    
+    
+})
 
-    it('does nothing when prompt is cancelled', async function () {
-        await uploadFileCommand(node, statFile, new FakeWindow(), new FakeCommands())
+describe('getFileToUpload', function () {
+    const fileLocation = vscode.Uri.file('/file.jpg')    
+    let window: FakeWindow
 
-        verify(s3.uploadFile(anything())).never()
+    const selection: any = {label: "Browse for more files..." }
+    const prompt: <T extends vscode.QuickPickItem>(opts: {
+        picker: vscode.QuickPick<T>
+        onDidTriggerButton?(
+            button: vscode.QuickInputButton,
+            resolve: (value: T[] | PromiseLike<T[] | undefined> | undefined) => void,
+            reject: (reason?: any) => void
+        ): void
+    }) => Promise<T[] | undefined> = () => {
+        return new Promise((resolve, reject) => {
+            resolve([selection])
+        })
+    }
+
+    beforeEach(function () {
+        window = new FakeWindow({ dialog: { openSelections: [fileLocation] } })       
+    })
+    
+    it("directly asks user for file if no active editor", async function () {
+        
+        const response = await getFileToUpload(undefined, window, prompt)
+        assert.strictEqual(response, fileLocation)
     })
 
-    it('shows an error message when upload fails', async function () {
-        when(s3.uploadFile(anything())).thenReject(new Error('Expected failure'))
+    it("Returns undefined if no file is selected on first prompt", async function () {
+        window = new FakeWindow({ dialog: { openSelections: undefined } })
+        
+        const response = await getFileToUpload(undefined, window, prompt)
+        assert.strictEqual(response, undefined)
+    })
 
-        const window = new FakeWindow({ dialog: { openSelections: [fileLocation] } })
-        const commands = new FakeCommands()
-        await uploadFileCommand(node, statFile, window, commands)
+    it("prompts quick pick with open file option, and browse for more files, returns open file if that option is selected", async function(){
+        selection.label = fileLocation.fsPath
+        const response = await getFileToUpload(fileLocation, window, prompt)
 
-        assert.ok(window.message.error?.includes('Failed to upload file'))
+        assert.strictEqual(response, fileLocation)
```

Comment:
For the sake of differentiating this test from the last test, use a different value for the fileLocation. Both this test and the next one make the same assertion and it can be difficult to see if the correct path is being tested since both pass in `fileLocation` through the `window` instead of the selection.

Try something like 
```
        const alreadyOpenedUri = new vscode.Uri('/alreadyOpened.txt')
        selection.label = alreadyOpenedUri.fsPath
        const response = await getFileToUpload(fileLocation, window, prompt)
        assert.strictEqual(response, alreadyOpenedUri)
```

This specifically passes both the `fileLocation` (via the FakeWindow) and a different value from the selection (`alreadyOpenedUri`).